Tests/NIOTests/FileRegionTest.swift

```diff
@@ -216,13 +216,13 @@ class FileRegionTest : XCTestCase {
                 let r = try Posix.read(descriptor: fd, pointer: &fr2Bytes, size: 5)
                 XCTAssertEqual(r, IOResult<Int>.processed(5))
             }
-            XCTAssertEqual(Array("01234".utf8), fr1Bytes)
-            XCTAssertEqual(Array("56789".utf8), fr2Bytes)
-
             defer {
                 // fr2's underlying fd must be closed by us.
                 XCTAssertNoThrow(try fh2.close())
             }
+
+            XCTAssertEqual(Array("01234".utf8), fr1Bytes)
```

Comment:
@weissi any reason why this was moved ?

Reply:
@normanmaurer yes, previously it didn't actually do anything in case of an error thrown. `defer`s are only executed on errors if the `defer` is syntactically before the error thrown. So generally cleanup-`defer`s need to go straight after the resource creation.

The previous one raised a warning in Swift 5